docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,378 @@
+
+```{note}
+If $\mathcal{F}$ is the fidelity defining the success of preparing an arbitrary pure state $\rho$, then
+$1-\mathcal{F}$ is the **average gate infidelity**.
+```
+
+For example, a depolarizing noise channel is a stochastic noise channel where a noiseless process is probabilistically mixed with orthogonal errors. If $\rho$ is a single qubit state, $p$ is the probabilistic error rate and $\epsilon(\rho)$ is the noise channel:
+
+```{math}
+\epsilon(\rho) = (1-p) \rho + p \frac{I}{2}
+```
+$\frac{I}{2}$ is the maximally mixed state which can be described using Paulis. 
+
+```{math}
+\frac{I}{2} = \frac{1}{4} (\rho + X \rho X + Y \rho Y + Z \rho Z)
+```
+
+Thus, the depolarizing channel can be redescribed using Paulis as shown below. 
+
+```{math}
+\epsilon(\rho) = (1-\frac{3p}{4}) \rho + \frac{p}{4} (X \rho X + Y \rho Y + Z \rho Z)
+
+```
+
+### Pauli Transfer Matrix
+
+Let $\Lambda(rho)$ be a $n-$qubit noise channel where $K_i$ are the Kraus operators.
+
+```{math}
+
+\Lambda(\rho) = \sum_{i=1} K_i \rho {K_i}^\dagger
+```
+If $P_i$ and $P_j$ are $n-$qubit Paulis, the following expression defines the entries of a Pauli Transfer Matrix (PTM). Here, 
+$i$ defines the rows while $j$ defines the columns of the PTM. All entries of the PTM are real and in the interval $[-1, 1]$.
+
+
+```{math}
+(R_{\Lambda})_{ij} = \frac{1}{2^n} \text{Tr} \{ P_i \Lambda(P_j)\}
+```
+A PTM allows us to distinguish between the two types of noise. The off-diagonal terms of the PTM are due to the effect of coherent noise while the diagonal terms are due to incoherent noise. PTM of a circuit is a product of the PTM of each layer in the circuit. Due to these, it is straightforward to see how coherent noise carries across different layers in the circuit and how incoherent errors are easier to deal with in the small error limit. 
+
+The known fault tolerant thresholds for stochastic noise are higher than coherent noise which makes the former a 'preferable' type of noise compared to the latter. If we want to not deal with coherent noise, Pauli twirling can be used to tailor coherent noise to incoherent noise. Same as the expression above, when a coherent noise channel is Pauli twirled, the noise channel can be described using Paulis after averaging over multiple Pauli twirled circuits. Refer to the [Pauli Twirling user guide](../guide/pt.md) for additional information. 
+
+## Using Pauli Twirling in Mitiq
+
+```{code-cell} ipython3
+# Utility functions for Pauli Twirling.
+
+import cirq
+import numpy as np
+import numpy.typing as npt
+from cirq.circuits import Circuit
+
+from mitiq.pec.channels import _circuit_to_choi, choi_to_super
+from mitiq.utils import matrix_to_vector, vector_to_matrix
+
+pauli_unitary_list = [
+    cirq.unitary((cirq.I)),
+    cirq.unitary((cirq.X)),
+    cirq.unitary((cirq.Y)),
+    cirq.unitary((cirq.Z)),
+]
+
+
+def _n_qubit_paulis(num_qubits: int) -> list[npt.NDArray[np.complex64]]:
```

Comment:
Lets not make these functions "private" since we are showing them to the reader.

It is possible to hide code cells, thought. Do you think we should make this code cell foldable so only the interested reader takes a look?